docs/jetstream.md

```diff
@@ -11,39 +11,107 @@ There are two components to this implementation:
 
 ### Controller
 
-The controller reconciles `NatsJetStreamChannel` CRDs by checking the availability of the `jetstream-channel-dispatcher`
-and ensures a `Service` corresponding to the channel exists with an `externalName` pointing to the 
-`jetstream-channel-dispatcher`.
+The controller reconciles `NatsJetStreamChannel` CRDs by ensuring the `Deployment` named `jetstream-ch-dispatcher`
+exists and ensures a `Service` corresponding to the channel exists with an `externalName` pointing to the
+`jetstream-ch-dispatcher`.
+
+#### Scoping
+
+`NatsJetStreamChannel` resources may be scoped to a namespace by adding the annotation
+`eventing.knative.dev/scope: namespace`, allowing the use of dedicated dispatcher deployments per namespace. This has
+two main benefits:
+
+- Channels across different namespaces cannot be starved of resources based on channels in other namespaces.
+- Channels across different namespaces can be configured with different credentials.
+
+#### Controller Dispatcher Configuration
+
+The controller dispatcher configuration tells the controller how dispatchers should be managed. It is not configuration
+for the dispatcher itself. This is implemented as a `ConfigMap` which is volume-mounted into the controller pod and must
+contain a file `jetstream-dispatcher-config`.
+
+In the below example, you may already have a `ConfigMap` named `config-jetstream` in `my-namespace` which serves another
```

Comment:
It's better to put `ConfigMap`   `config-jetstream` in ns `knative-eventing`. And If `namespaceDefaults` be set, then use the configmap in the specific namespace, such as below example, use the `configmap` `my-namespace/custom-jetstream-config`. If not , then just use the `clusterDefault`

Reply:
Yeah, I think we mean the same thing.

- There's a config map `knative-eventing/config-jetstream` which is used for the `knative-eventing/jetstream-ch-dispatcher` deployment.
- If a dispatcher is created in another namespace:
  - If there's a namespaceDefaults value for the target namespace, use that config map.
  - Otherwise It should fallback to use the clusterDefault config map

If you agree, the main problem now is if the `my-namespace/jetstream-ch-dispatcher` is using the `clusterDefault` config map, how to get that data into `my-namespace` because you can't mount config maps cross-namespace. Possible solutions:

1. The controller creates a config map of that name in `my-namespace` and volume-mounts the object onto the dispatcher.
2. The dispatcher reads the config map from the `knative-eventing` namespace manually and no volume mounts occur. This would require configuring extra roles for the `ServiceAccount` the dispatcher runs as.

Reply:
I think option 2 is better, because the `ConfigMap` also references `Secret` objects for credentials. If we copied the config we'd also have to copy the secret.

With option 2 you can just give the dispatcher `ServiceAccount` the correct roles and no other workload in the namespace will be allowed to read them.

Reply:
Actually, I'm making an issue out of something which isn't one...

We can just say that if you create a scoped channel, it's your responsibility to ensure the `ConfigMap` (and any `Secret` resources which are referenced) are created in that namespace.

Reply:
Option 2 looks good to me. But, another thing, Who will create the deployment `my-namespace/jetstream-ch-dispatcher`? Created mannully or by controller？


Can this problem be solved by maintaining several connections（for diferrent connection config）in the `knative-eventing/jetstream-ch-dispatcher` deployment？